client/src/api/v2/camera.js

```diff
@@ -39,3 +41,117 @@ export function useOverlays(device) {
 
   return { overlays, setActiveOverlay };
 }
+
+/**
+ * Send a message to server to start recording video
+ */
+export function startRecording() {
+  emit('start-camera-recording');
+}
+
+/**
+ * Send a message to server to stop recording video
+ */
+export function stopRecording() {
+  emit('stop-camera-recording');
+}
+
+/**
+ * @typedef {object} CameraRecordingStatusPayload
+ * @property {string} status Status of recording
+ * @property {?string} recordingMinutes Length of recording
+ * @property {?string} recordingFile File name of recording
+ * @property {?string} diskSpaceRemaining Remaining space on disk
+ * @property {?string} error Error message from camera
+ */
+
+/**
+ * Parse the payload into an object and format the values
+ *
+ * Payload structure is defined in the 'V3 MQTT Topics' page on Notion.
+ * Topic is /v3/camera/recording/status/<primary/>secondary>.
+ *
+ * @param {string} payload Payload
+ * @returns {CameraRecordingStatusPayload} Formatted payload without status
+ */
+function parsePayload(payload) {
+  const data = typeof payload === 'string' ? JSON.parse(payload) : payload;
+  if (!data) return null;
+
+  return Object.keys(data)
+    .reduce((acc, field) => {
```

Comment:
I'm not a fan of reduce... In my opinion a for-loop would give more clarity.

Reply:
Changed `reduce` to `forEach`
9c115e8